R/save.r

```diff
@@ -58,6 +60,25 @@ ggsave <- function(filename, plot = last_plot(),
   invisible()
 }
 
+#' Parse a DPI input from the user
+#'
+#' Allows handling of special strings when user specifies a DPI like "print".
+#'
+#' @param dpi Input value from user
+#' @return Parsed DPI input value
+#' @noRd
+parse_dpi <- function(dpi) {
+  if (is.character(dpi)) {
+    dpi <- switch(dpi, screen = 72, print = 300, retina = 320, NULL)
```

Comment:
Instead of NULL, you should insert the error message. i.e.

```R
dpi <- switch(dpi, 
  screen = 72, 
  print = 300, 
  retina = 320,
  stop("Unknown DPI string", call. = FALSE)
)